### PR TITLE
src/utils cmd/lachesis : add pidfile check

### PR DESCRIPTION
**Pull request by Maxime2**

### CODE REVIEW

File: src/utils/checkpid.go [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+import (
+	"fmt"
+	"os"
+	"syscall"
+	"github.com/facebookgo/pidfile"
+)
+
+func CheckPid() error {
+	pidfile.SetPidfilePath("/tmp/lachesis.pid")
```
Comment (SamuelMarks) [inline src/utils/checkpid.go]: Can we choose a better location for this, something standardised?

Comment (Maxime2) [reply src/utils/checkpid.go]: /var/run is write protected for ordinary users, so we need do some research on better practice.

File: cmd/lachesis/commands/run_lachesis.go [hunk 2]
```diff
@@ -7,10 +7,16 @@ package commands
 import (
 	"github.com/spf13/cobra"
 	"github.com/spf13/viper"
+	"github.com/Fantom-foundation/go-lachesis/src/utils"
 )
 
 func runLachesis(cmd *cobra.Command, args []string) error {
 
+	err := utils.CheckPid()
```
Comment (SamuelMarks) [inline cmd/lachesis/commands/run_lachesis.go]: Surround it with this? - `if runtime.GOOS != "windows" {`

File: src/utils/checkpid.go [hunk 3]
```diff
@@ -0,0 +1,30 @@
+package utils
+
+import (
+	"fmt"
+	"os"
+	"syscall"
+	"github.com/facebookgo/pidfile"
```
Comment (SamuelMarks) [inline src/utils/checkpid.go]: This hasn't been updated for 3 years. Are we sure this is a good idea?

File: src/utils/checkpid.go (hunk 1 above)
Comment (SamuelMarks) [reply src/utils/checkpid.go]: Then expose a `--pidfile` CLI argument, with a default of `/tmp/go-lachesis.pid`. The main use-case for introducing PID files, is so I can integrate it cleanly into systemd; distributing with .deb.

File: src/utils/checkpid.go (hunk 3 above)
Comment (Maxime2) [reply src/utils/checkpid.go]: I don't think anything need to be added into this, it just does what it's supposed to do.

File: src/utils/checkpid.go (hunk 1 above)
Comment (Maxime2) [reply src/utils/checkpid.go]: Added --pidfile CLI argument.

File: cmd/lachesis/commands/run_lachesis.go (hunk 2 above)
Comment (Maxime2) [reply cmd/lachesis/commands/run_lachesis.go]: Surrounded.

**Pull request merged**